libstormweaver/src/metadata.cpp

```diff
@@ -1,10 +1,50 @@
 
 #include "metadata.hpp"
 
+#include <algorithm>
+#include <fmt/format.h>
 #include <iostream>
 
 namespace metadata {
 
+bool Index::operator==(const Index &other) const {
+  if (name != other.name || unique != other.unique ||
+      fields.size() != other.fields.size()) {
+    return false;
+  }
+
+  return fields == other.fields;
+}
+
+bool Index::operator!=(const Index &other) const { return !(*this == other); }
+
+bool Table::operator==(const Table &other) const {
+  if (name != other.name || engine != other.engine ||
+      tablespace != other.tablespace || partitioning != other.partitioning ||
+      columns.size() != other.columns.size() ||
+      indexes.size() != other.indexes.size()) {
+    return false;
+  }
+
+  for (const auto &column : columns) {
+    if (std::find(other.columns.begin(), other.columns.end(), column) ==
+        other.columns.end()) {
+      return false;
+    }
+  }
```

Comment:
can't we just use?
```cpp
if (columns != other.columns) {return false;}
```

Reply:
That would be order dependent, while this is order independent. The order of the columns doesn't matter, and two parallel workers altering could cause an inconsistency between the ordering in the database and the metadata.